pkg/shared/util/scram_client.go

```diff
@@ -0,0 +1,38 @@
+// copied from https://github.com/IBM/sarama/blob/main/examples/sasl_scram_client/scram_client.go
+package util
```

Comment:
Maybe combining this file with sasl_config.go, since it's only used for that at this moment?